pyfar/dsp/fft.py

```diff
@@ -301,16 +404,101 @@ def _n_bins(n_samples):
     ---------
     n_samples : int
         Number of samples
+    complex : bool
+        Flag which indicates if the time data are real or complex-valued.
 
     Returns
     -------
     n_bins : int
         Resulting number of frequency bins
 
     """
-    if _is_odd(n_samples):
-        n_bins = (n_samples+1)/2
+    if complex:
+        n_bins = n_samples
     else:
-        n_bins = n_samples/2+1
+        n_bins = n_samples // 2 + 1
 
     return int(n_bins)
+
+
+def _calc_n_samples_from_frequency_data(num_freq_bins, complex=False):
+    """
+    Helper function to calculate the number of samples resulting from
+    an inverse FFT of a spectrum with n_freq_bins
+
+    Paramters
+    ---------
+    num_freq_bins : int
+        Number of frequency bins
+    complex : bool
+        Flag which indicates if the frequency data are a one or two-sided
+        spectrum.
+
+    Returns
+    -------
+    n_samples : int
+        Resulting number of time samples.
+
+    """
+    if complex:
+        return num_freq_bins
+    else:
+        return max(1, (num_freq_bins - 1) * 2)
+
+
+def add_mirror_spectrum(data_single_sided, even):
+    """
+    Adds mirror spectrum to single-sided frequency data
+    and applies fftshift. The output is a double-sided
+    spectrum that matches the format of :py:func:`~fft`.
+
+    Paramters
+    ---------
+    data : numpy array
+        M-dimensional array of single-sided spectrum of shape (..., N)
+        containing N frequency bins.
+    even : bool
+        flag which indicates if the number of sampels of the time
+        data were even.
+
+    Returns
+    -------
+    data : numpy array
+        M-dimensional array of double-sided spectrum of shape (..., N)
+        containing N frequency bins.
+
+    """
+    if even:
+        mirror_spec = data_single_sided[..., 1:-1]
+    else:
+        mirror_spec = data_single_sided[..., 1:]
+
+    mirror_spec = np.conj(np.flip(mirror_spec, axis=-1))
+    data = np.concatenate((data_single_sided, mirror_spec), axis=-1)
+    data[..., 0] = np.real(data[..., 0])  # ensure DC bin is real valued
+    return sfft.fftshift(data, axes=-1)
+
+
+def remove_mirror_spectrum(data_double_sided):
```

Comment:
I noted that we have code for checking if a spectrum is symmetric. Why not put this here before removing the symmetric part?

Reply:
Maybe I am wrong, but in case we want to change the complex flag from True to False for frequency domain signals, we first need to check if we are allowed to kill the mirror spectrum (_check_conjugate_symmetry()), and then kill it? Or did I miss your point ?

Reply:
I have now shifted the test for conjugate symmetry to `remove mirror_spec`